src/Core/Types/Unifier.cs

```diff
@@ -581,7 +588,44 @@ private bool TryMakeFieldName(StructureField fa, StructureField fb, out string n
             return true;
         }
 
-		public DataType UnifyPointer(Pointer ptrA, DataType b)
+        private bool IsNestedStructure(
+            StructureField fa, StructureField fb,
+            out StructureField fNestedStruct, out StructureField fOther)
+        {
+            var strFa = fa.DataType.TypeReferenceAs<StructureType>();
+            var strFb = fb.DataType.TypeReferenceAs<StructureType>();
+            fNestedStruct = null;
+            fOther = null;
+            if (
+                (strFa == null && strFb == null) ||
+                (strFa != null && strFb != null))
+                return false;
+
+            var strSize = ((strFa != null) ? strFa : strFb).GetInferredSize();
+            fNestedStruct = (strFa != null) ? fa : fb;
+            fOther = (strFb != null) ? fa : fb;
```

Comment:
`strFb != null` implies `strFa == null` (see line 599). Why not change this to 
`fOther = (strFa != null) ? fb : fa;` or even better:
```
if (strFa != null) {
    fNestedStruct = fa;
    fOther = fb;
} else {
    fNestedStruct = fb;
   fOther = fa;
}
var strSize = fNestedStruct.GetInferredSize();
```
Although a little longer, the logic is clearer and you avoid the repetition of the comparison three times.